// thirdweb/dist/cjs/wallets/in-app/core/authentication/guest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.guestAuthenticate = guestAuthenticate;
const fetch_js_1 = require("../../../../utils/fetch.js");
const getLoginPath_js_1 = require("./getLoginPath.js");
/**
 * Does no real authentication, just issues a temporary token for the user.
 * @internal
 */
async function guestAuthenticate(args) {
    const clientFetch = (0, fetch_js_1.getClientFetch)(args.client, args.ecosystem);
    const authResult = await (async () => {
        const path = (0, getLoginPath_js_1.getLoginCallbackUrl)({
            authOption: "guest",
            client: args.client,
            ecosystem: args.ecosystem,
        });
        const res = await clientFetch(`${path}`, {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify({
                sessionId: "guest",
            }),
        });
        if (!res.ok)
            throw new Error("Failed to generate guest account");
        return (await res.json());
    })();
    return authResult;
}
